src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.RegularCompilationTracker_Generators.cs

```diff
@@ -282,7 +282,7 @@ private partial class RegularCompilationTracker : ICompilationTracker
             if (compilationWithStaleGeneratedTrees != null)
             {
                 var generatedTreeCount =
-                    runResult.Results.Sum(r => IsGeneratorRunResultToIgnore(r) ? 0 : r.GeneratedSources.Length);
+                    runResult.Results.Sum(r => r.GeneratedSources.IsDefaultOrEmpty ? 0 : r.GeneratedSources.Length);
```

Comment:
This is another case of the bug that was fixed here: https://github.com/dotnet/roslyn/pull/74507